rust/src/client.rs

```diff
@@ -0,0 +1,842 @@
+use std::{
+    convert::TryInto,
+    ffi::{self, CStr, CString},
+    os::raw::{c_char, c_void},
+    ptr, slice,
+    time::Duration,
+};
+
+use crossbeam_utils::thread;
+use libsettings_sys::{
+    sbp_msg_callback_t, sbp_msg_callbacks_node_t, settings_api_t, settings_create,
+    settings_destroy, settings_read_bool, settings_read_by_idx, settings_read_float,
+    settings_read_int, settings_read_str, settings_t,
+    settings_write_res_e_SETTINGS_WR_MODIFY_DISABLED, settings_write_res_e_SETTINGS_WR_OK,
+    settings_write_res_e_SETTINGS_WR_PARSE_FAILED, settings_write_res_e_SETTINGS_WR_READ_ONLY,
+    settings_write_res_e_SETTINGS_WR_SERVICE_FAILED,
+    settings_write_res_e_SETTINGS_WR_SETTING_REJECTED, settings_write_res_e_SETTINGS_WR_TIMEOUT,
+    settings_write_res_e_SETTINGS_WR_VALUE_REJECTED, settings_write_str, size_t,
+};
+use log::{debug, error};
+use sbp::{
+    link::{Key, Link},
+    messages::{SBPMessage, SBP},
+    serialize::SbpSerialize,
+};
+
+use crate::{settings, SettingKind, SettingValue};
+
+const SENDER_ID: u16 = 66;
```

Comment:
Just so it's clear that this is the same sender ID as the other settings clients:

```suggestion
const SENDER_ID: u16 = 0x42;
```